File: lib/java/src/protocol/TProtocolFactory.java
package com.facebook.thrift.protocol;

import com.facebook.thrift.transport.TTransport;

/**
 * Factory interface for constructing protocol instances.
 *
 * @author Mark Slee <mcslee@facebook.com>
 * @author Aditya Agarwal <aditya@facebook.com>
 */
public interface TProtocolFactory {
  public TProtocol getProtocol(TTransport trans);
}

File: lib/csharp/src/Protocol/TProtocolFactory.cs
//
//  TProtocolFactory.cs
//
//  Begin:  Dec 3, 2007
//  Authors:
//		Will Palmeri <wpalmeri@imeem.com>
//
//  Distributed under the Thrift Software License
//
//  See accompanying file LICENSE or visit the Thrift site at:
//  http://developers.facebook.com/thrift/using
using System;
using System.Collections.Generic;
using System.Text;
using Thrift.Transport;

namespace Thrift.Protocol
{
	public interface TProtocolFactory
	{
		TProtocol GetProtocol(TTransport trans);
	}
}
